### DISCUSSION TITLE
Pyright settings that disable Union type warnings

### DISCUSSION THREAD

**Discussion by edwardjjj** [microsoft/pyright#7452] [Q&A] [answered]
# Problem Description

I'm working with a machine learning library where a function returns Union types. Pyright by default showing diagnostics about Union types whereasPylance doesn't. What settings should I use to get the same behavior?
# Environment
Neovim: 0.9.5
Python: 3.9
Pyright: 1.1.353
# Code snapshots
## pyright
![image](https://github.com/microsoft/pyright/assets/76624665/a16dc731-7892-4940-8d67-b24d4f51e9c4)
## pylance
![image](https://github.com/microsoft/pyright/assets/76624665/cfbe96bc-18ca-47b6-a78b-8ea02b1903cb)

**Comment by erictraut** [comment, marked as answer]
If you're seeing different behavior between pyright and pylance, it's because of one of the reasons documented [here](https://github.com/microsoft/pylance-release/blob/main/USING_WITH_PYRIGHT.md).

**Reply by edwardjjj** [reply]
Thank you for pointing me to the docs. I just realized that type-checking is disabled by default on Pylance. Now I'm getting the same results.
